### PR TITLE
 Expose tick_height to bpf programs

### PR DESCRIPTION
**Pull request by mvines**
BPF programs now receive the current height of the cluster.

Fixes: #1359

### CODE REVIEW

File: programs/native/bpf_loader/src/lib.rs [hunk 1]
```diff
@@ -88,13 +88,18 @@ pub fn helper_sol_log_u64(arg1: u64, arg2: u64, arg3: u64, arg4: u64, arg5: u64)
     0
 }
 
-fn create_vm(prog: &[u8]) -> Result<EbpfVmRaw, Error> {
+pub fn helper_sol_tick_height(_arg1: u64, _arg2: u64, _arg3: u64, _arg4: u64, _arg5: u64) -> u64 {
+    0 // TODO!  Get tick_height somehow...
```
Comment (mvines) [inline programs/native/bpf_loader/src/lib.rs]: Right now Ebpf helper functions can have no context as they are just function pointers.  I think I'd like to modify Ebpf to take a Fn trait instead so we can have closures.  @jackcmay wdyt?

Comment (mvines) [reply programs/native/bpf_loader/src/lib.rs]: I went with option 2, extending `sol_deserialize()`

Comment (jackcmay) [reply programs/native/bpf_loader/src/lib.rs]: I'll be digging into the jit implementation as part of loop caps and as part of monetizing instructions.  Will look at adding closures then.  For now, passing in tick_hight seems ok.

File: programs/native/bpf_loader/src/lib.rs [hunk 2]
```diff
@@ -7,6 +7,7 @@ extern crate env_logger;
 extern crate log;
 extern crate libc;
 extern crate solana_rbpf;
+#[macro_use]
```
Comment (jackcmay) [inline programs/native/bpf_loader/src/lib.rs]: I thought this went away in a recent release of rust, Is macro_use still required?

Comment (mvines) [reply programs/native/bpf_loader/src/lib.rs]: Removing it produces:
```
error: cannot find macro `solana_entrypoint!` in this scope                                                                                        
   --> programs/native/bpf_loader/src/lib.rs:142:1                                                                                                 
    |                                                                                                                                              
142 | solana_entrypoint!(entrypoint);                                                                                                              
    | ^^^^^^^^^^^^^^^^^                 
```

**Comment by rob-solana** [comment]
approved

**Pull request merged**